static/js/offline-search.js

```diff
@@ -0,0 +1,104 @@
+// Adapted from code by Matt Walters https://www.mattwalters.net/posts/hugo-and-lunr/
+
+var idx = null;         // Lunr index
+var resultDetails = []; // Will hold the data for the search results (titles and summaries)
+var $searchInput;       // The search box element in the navbar
+var $searchResults;     // Results shown in the navbar
+
+$(window).on('load', function() {
+    // Set up for an Ajax call to request the JSON data file that is created by
+    // Hugo's build process, with the template we added above
+    var request = new XMLHttpRequest();
+    var query = '';
+
+    // Get dom objects for the elements we'll be interacting with
+    $searchResults = document.getElementById('search-results');
+    $searchInput   = document.getElementById('search-input');
+
+    request.overrideMimeType("application/json");
+    request.open("GET", "/index.json", true); // Request the JSON file created during build
+    request.onload = function() {
+    if (request.status >= 200 && request.status < 400) {
+        // Success response received in requesting the index.json file
+        var searchDocuments = JSON.parse(request.responseText);
+
+        // Build the index so Lunr can search it.  The `ref` field will hold the URL
+        // to the page/post.  title, excerpt, and body will be fields searched.
+        idx = lunr(function lunrIndex() {
+        this.ref('ref');
+        this.field('title');
+        this.field('excerpt');
```

Comment:
this.feild("excerpt") is unnecessary because this.field('body') contains all text.
```
- this.field('excerpt');
```